ext/xml/xml.c

```diff
@@ -1412,35 +1412,54 @@ PHP_FUNCTION(xml_parser_free)
 PHP_FUNCTION(xml_parser_set_option)
 {
 	xml_parser *parser;
-	zval *pind, *val;
-	zend_long opt;
+	zval *pind;
+	zend_long opt, val_long;
+	zend_string *val_str;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Olz", &pind, xml_parser_ce, &opt, &val) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(3, 3)
+    	Z_PARAM_OBJECT_OF_CLASS(pind, xml_parser_ce)
+    	Z_PARAM_LONG(opt)
+    	Z_PARAM_STR_OR_LONG(val_str, val_long)
+    ZEND_PARSE_PARAMETERS_END();
```

Comment:
Same here.

Reply:
I'll change this too if the newly introduced approach is appropriate.